### PR TITLE
Typehints and more math funcs for units

### LINKED ISSUE #2062: Include Unit in type annotations
Functions like `sc.sqrt` can work with units but that is not reflected int he type annotations.

Also make sure that all maths functions actually support units.

### PR DESCRIPTION
**Pull request by jl-wynen**
Fixes #2062

### CODE REVIEW

File: tests/core/math_test.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+def test_pow():
+    assert_export(sc.pow, sc.scalar(0.0), sc.scalar(0.0))
+    assert_export(sc.pow, sc.scalar(0.0), 0.0)
+    assert_export(sc.pow, sc.units.one, 0.0)
+
+
+def test_reciprocal():
+    assert_export(sc.reciprocal, sc.scalar(0.0))
+    assert_export(sc.reciprocal, sc.units.one)
```
Comment (nvaytet) [inline tests/core/math_test.py:65]: In this particular case, you are asserting the export, but in the python bindings, we are not simply forwarding to the C++ function, we are computing the reciprocal there directly. So just checking if the function is exported isn't enough, i think?

File: tests/core/math_test.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+
+def test_norm():
+    assert_export(sc.norm, sc.vector(value=[0.0, 0.0, 0.0]))
+
+
+def test_pow():
+    assert_export(sc.pow, sc.scalar(0.0), sc.scalar(0.0))
+    assert_export(sc.pow, sc.scalar(0.0), 0.0)
+    assert_export(sc.pow, sc.units.one, 0.0)
```
Comment (nvaytet) [inline tests/core/math_test.py:60]: Missing assert for `int64`?

File: tests/core/math_test.py (hunk 1 above)
Comment (jl-wynen) [reply tests/core/math_test.py]: :+1: There was actually a bug in the bindings.

Comment (nvaytet) [reply tests/core/math_test.py]: The power of unit tests :+1:

**Review by nvaytet — APPROVED** [review]

**Pull request merged**